Neos.ContentRepository.Core/Classes/Subscription/SubscriptionStatuses.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\Subscription;
+
+/**
+ * A collection of the states of the subscribers.
+ *
+ * Currently only projections are the available subscribers, but when the concept is extended,
+ * other *SubscriptionStatus value objects will also be hold in this set.
+ * Like "ListeningSubscriptionStatus" if a "ListeningSubscriber" is introduced.
+ *
+ * In case the subscriber is not available currently - e.g. will be detached, a {@see DetachedSubscriptionStatus} will be returned.
+ * Note that ProjectionSubscriptionStatus with status == Detached can be returned, if the projection is installed again!
+ *
+ * @api
+ * @implements \IteratorAggregate<ProjectionSubscriptionStatus|DetachedSubscriptionStatus>
+ */
+final readonly class SubscriptionStatuses implements \IteratorAggregate
```

Comment:
there is a little naming conflict, this is NOT the collection of the `SubscriptionStatus` enum but there is one layer of objects in between ...

Reply:
name to `subscriptionStatus()`  and `SubscriptionStatusCollection`